flycheck.el

```diff
@@ -6558,6 +6561,56 @@ See URL `https://docs.python.org/3.4/library/py_compile.html'."
           line ", " column ", " (one-or-more not-newline) line-end))
   :modes python-mode)
 
+(flycheck-define-checker r-lint
+  "An R style checker using the lint package.
+
+See URL `http://cran.r-project.org/web/packages/lint/'."
+  :command ("R" "--slave" "--restore" "--no-save" "-e"
+            (eval (concat
+                   "library(lint);"
+                   "try(lint(commandArgs(TRUE), lint.style))"))
+            "--args" source)
+  :error-patterns
+  ((info line-start "Lint: " (minimal-match (message))
+         ": found on lines " line (optional (one-or-more ", " line))
+         line-end))
+  :modes ess-mode
+  :next-checkers (r-lintr))
+
+(flycheck-define-checker r-lintr
+  "An R style and syntax checker using the lintr package.
+
+See URL `http://cran.r-project.org/web/packages/lintr/'."
+  :command ("R" "--slave" "--restore" "--no-save" "-e"
+            (eval (concat
+                   "library(lintr);"
+                   "try(lint(commandArgs(TRUE), default_linters))"))
+            "--args" source)
+  :error-patterns
+  ((info line-start (file-name) ":" line ":" column ": style: " (message)
+         line-end)
+   (warning line-start (file-name) ":" line ":" column ": warning: " (message)
+            line-end)
+   (error line-start (file-name) ":" line ":" column ": error: " (message)
+          line-end))
+  :modes ess-mode
+  :next-checkers ((warning . r-svtools)))
+
+(flycheck-define-checker r-svtools
```

Comment:
@dholm Can you explain the difference between `svtools` and `lintr`?  The svtools site isn't very informative about the purpose and the features of this package.  Also, it doesn't seem to be well-maintained either:  svtools seems to be a side-kick of a GUI library, rather than a fully-fledged independent linter.